CMakeLists.txt

```diff
@@ -1,11 +1,144 @@
 cmake_minimum_required(VERSION 3.9)
-project(readerwriterqueue VERSION 1.0.0)
+set(CMAKE_CXX_STANDARD 11)  # don't need to specify manually per target anymore.
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 include(GNUInstallDirs)
 
-add_library(${PROJECT_NAME} INTERFACE)
+project(readerwriterqueue)
 
-target_include_directories(readerwriterqueue INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
+add_library(readerwriterqueue INTERFACE)
 
-install(FILES atomicops.h readerwriterqueue.h readerwritercircularbuffer.h LICENSE.md
+# This needs to be updated everytime the library tag version updates. The version is not 1.0.0, it's 1.0.6 according to tags.
+set_target_properties(readerwriterqueue PROPERTIES
+        SOVERSION 1
+        VERSION 1.0.6)
```

Comment:
Is this physically related to the GitHub release version? Or independent?

Reply:
This relates to actual software compatibility, which should normally be 1:1 to the Git tag version.  There's no official automated tools to grab this, but I typically manually edit it in since it's so simple to do. I assumed you were using semver, if that's not the case that's a whole new can of worms to deal with since I believe the current VCPKG port of this assumes semver.   SO version is ABI incompatibility, and typically should be the same as the major version on VERSION, underneath though I believe only windows uses that.